src/sth_database.js

```diff
@@ -330,38 +338,107 @@
-                // We have observed that although leaving the strict option to the default value, sometimes
-                //  we get a 'collection already exists' error when executing connection.db#createCollection()
-                connection.db.collection(collectionName, {strict: true},
-                  function (err, collection) {
-                    return callback(err, collection);
+                    }
+                  );
+                }
+              }
+            );
+          }
+        } else {
+          collection.ensureIndex(
+            {
+              '_id.origin': 1
+            },
+            {
+              expireAfterSeconds: parseInt(sthConfig.TRUNCATION_EXPIREAFTERSECONDS)
+            }, function(err) {
+              if (err) {
+                sthLogger.error(
+                  'Error when creating the index for TTL for collection \'' +
+                  collectionName + '\': ' + err,
+                  {
+                    operationType: sthConfig.OPERATION_TYPE.DB_LOG
                   }
                 );
-              } else {
-                return callback(err, collection);
               }
             }
           );
+        }
+      }
+    };
+
+    var createCollectionCB = function (err, collection) {
+      if (!err && shouldStoreHash) {
+        storeCollectionHash(params, isAggregated, collectionName, function(hashErr) {
+          if (hashErr) {
+            // There was an error when storing the collection hash
+            // Do nothing
+            sthLogger.warn('Error when storing the hash generated as part of the collection name into the database', {
+              operationType: sthConfig.OPERATION_TYPE.DB_LOG
+            });
+          }
+          if (shouldTruncate) {
+            setTTLPolicy(collection);
+          }
+          return callback(hashErr, collection);
+        });
+      } else if (err && err.message === 'collection already exists') {
+        // We have observed that although leaving the strict option to the default value, sometimes
+        //  we get a 'collection already exists' error when executing connection.db#createCollection()
+        connection.db.collection(collectionName, {strict: true},
+          function (err, collection) {
+            return callback(err, collection);
+          }
+        );
+      } else {
+        if (shouldTruncate) {
+          setTTLPolicy(collection);
+        }
+        return callback(err, collection);
+      }
+    };
+
+    connection.db.collection(collectionName, {strict: true},
+      function (err, collection) {
+        if (err &&
+          (err.message === 'Collection ' + collectionName + ' does not exist. Currently in strict mode.') &&
+            shouldCreate) {
+          if (shouldTruncate && !isAggregated) {
+            // Set the size removal policy if required
+            if (parseInt(sthConfig.TRUNCATION_SIZE) > 0) {
```

Comment:
I'm wondering if some kind of input parameters validation is done. If that's the case, then is not necessary to ckeck for it here.